src/app/features/courses/courses.component.ts

```diff
@@ -9,10 +10,16 @@ import { Course } from './models/course';
 })
 export class CoursesComponent implements OnInit {
   public courses: Course[] = [];
+  public searchValue: string = '';
+  private filter = new FilterPipe();
```

Comment:
This is not the right way to use a filter pipe. You should use it via a template, do not create it manually:

```
    <app-courses-list
      *ngIf="courses.length; else not_found"
      [courses]="courses | filter:searchValue | orderBy"
    ></app-courses-list>

```

and overwrite the output value as follows:

```
  onSearch(value: string): void {
    this.searchValue = value;
  }
```